### PR TITLE
Add Server Groups functions in OpenStack driver

### PR DESCRIPTION
**Pull request by micafer**
### Description

Enable to list, get, create and delete server groups.
Also enable to set the server group in the create node function to enable affinity and anti-affinity policies.
#1629 

### Status

Replace this: describe the PR status. Examples:

- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [ ] Documentation
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Review by dpeschman — APPROVED** [review]

File: libcloud/compute/drivers/openstack.py [hunk 1]
```diff
@@ -4813,3 +4910,47 @@ def __repr__(self):
...
+    See:
+    https://docs.openstack.org/api-ref/compute/?expanded=create-server-detail,list-server-groups-detail#server-groups-os-server-groups
+    """
+
+    def __init__(
+        self,
+        id,
+        name,
+        policy,
+        members=[],
```
Comment (Kami) [inline libcloud/compute/drivers/openstack.py]: Should avoid mutable defaults here for ``members`` and ``rules``argument and default to ``None`` and set the value to empty list if it's not set inside the constructor (self.members = members or []``.

**Review by Kami — APPROVED** [review]
Besides the mutable default value issue in the constructor arguments, LGTM :+1:

**Comment by micafer** [comment]
Hi @Kami,
Integration Tests are failing and the error seems that is not related with this PR.
Any ideas?

**Comment by Kami** [comment]
@micafer Not sure why those tests are failing. Could be a temporary issue since it works fine in other PRs (so re-run may help).

**Review by Kami — APPROVED** [review]
LGTM

**Pull request merged**